src/types/polygon.js

```diff
@@ -4,8 +4,9 @@ import {setBorderStyle, resolvePointPosition, getElementCenterPoint} from '../he
 
 export default class PolygonAnnotation extends Element {
 
-  inRange(x, y) {
-    return this.vertices && this.vertices.length > 0 && pointIsInPolygon(this.vertices, x, y);
+  inRange(mouseX, mouseY, useFinalPosition) {
+    const {vertices} = this.getProps(['vertices'], useFinalPosition);
```

Comment:
This wont work. If the vertices are calculated in `draw`, we need to calculate them here, with the final x/y.

I think it could be better to calculate the points of the polygon in resolveElementProperties. Just need to make the animator animate those points. I guess it would be easiest if the points were handled as sub elements and their properties would be updated in similar manner (using updateElement)

We can just leave the Polygon annotation out of this pr, because its more complicated.

Reply:
Let me propose to apply the double calculattion pf vertices here and then submit another PR to move the calculation in resolveElementOptions. With double calculation the polygon is animated anyway.
What do you think?

Reply:
@kurkle as I wrote, I have implemented the vertices are not stored in the element anymore and in the `draw` (therefore there are animated by x,y center point and not the vertices coordinates, todo) and in `inRange`.